test/no-single-promise-in-promise-methods.mjs

```diff
@@ -0,0 +1,136 @@
+import outdent from 'outdent';
+import {getTester} from './utils/test.mjs';
+
+const {test} = getTester(import.meta);
+
+const error = {
+	messageId: 'no-single-promise-in-promise-methods',
+};
+
+test({
+	valid: [
+		outdent`
+			Promise.all([promise, anotherPromise])
+		`,
+		outdent`
+			Promise[all]([promise, anotherPromise])
+		`,
+		outdent`
+			Promise.all(notArrayLiteral)
+		`,
+		outdent`
+			Promise.all([...promises])
+		`,
+		outdent`
+			Promise.any([promise, anotherPromise])
+		`,
+		outdent`
+			Promise[any]([promise, anotherPromise])
+		`,
+		outdent`
+			Promise.any(notArrayLiteral)
+		`,
+		outdent`
+			Promise.any([...promises])
+		`,
+		outdent`
+			Promise.race([promise, anotherPromise])
+		`,
+		outdent`
+			Promise[race]([promise, anotherPromise])
+		`,
+		outdent`
+			Promise.race(notArrayLiteral)
+		`,
+		outdent`
+			Promise.race([...promises])
+		`,
+		outdent`
+			Promise.allSettled([promise])
+		`,
+	],
+
+	invalid: [
+		{
+			code: outdent`
+				Promise.all([promise])
+			`,
+			errors: [error],
+			output: outdent`
+				promise
+			`,
+		},
+		{
+			code: outdent`
+				Promise[all]([promise])
+			`,
+			errors: [error],
+			output: outdent`
+				promise
+			`,
+		},
+		{
+			code: outdent`
+				Promise.all([(promise, anotherPromise)])
+			`,
+			errors: [error],
+			output: outdent`
+				(promise, anotherPromise)
+			`,
+		},
+		{
+			code: outdent`
+				Promise.any([promise])
+			`,
+			errors: [error],
+			output: outdent`
+				promise
+			`,
+		},
+		{
+			code: outdent`
+				Promise[any]([promise])
+			`,
+			errors: [error],
+			output: outdent`
+				promise
+			`,
+		},
+		{
+			code: outdent`
+				Promise.any([(promise, anotherPromise)])
+			`,
+			errors: [error],
+			output: outdent`
+				(promise, anotherPromise)
+			`,
+		},
+		{
+			code: outdent`
+				Promise.race([promise])
+			`,
+			errors: [error],
+			output: outdent`
+				promise
+			`,
+		},
+		{
+			code: outdent`
+				Promise[race]([promise])
+			`,
+			errors: [error],
+			output: outdent`
+				promise
+			`,
+		},
+		{
+			code: outdent`
+				Promise.race([(promise, anotherPromise)])
+			`,
+			errors: [error],
+			output: outdent`
+				(promise, anotherPromise)
```

Comment:
This file will break

```js
foo
Promise.race([(promise, anotherPromise)])
```

Reply:
I have restricted the rule to CallExpression, Identifier and MemberExpression (await or not) to avoid this kind of case.

Reply:
The correct fix should check ASI problem instead, we have `./utils/needs-semicolon.js` can be used.

Reply:
I authorized all types and managed the special cases of SpreadElement and SequenceExpression.
I didn't need to use `./utils/needs-semicolon.js`.